### PR TITLE
feat: display data fields on graph

### PR DESCRIPTION
**Pull request by jessicamcinchak**
Cherry picked from our prior Biennale experimental / maximalist graph styling #2947 

Wondering if displaying data fields in the graph makes sense alongside preparing to launch search for "data fields only"? Eg, how familiar do we think average editors are with the concept of "data fields" ?

Just an idea, very much think this one is up to Daf, Ian and content team to decide if useful! While I like this style for increased visibility, it definitely stretches out overall size of graphs which may prove harder to navigate?

**Before:**
Screenshot from 2024-09-15 20-22-02

**After:**
Screenshot from 2024-09-15 20-16-58

** Does _not_ display data fields (yet?) for any of the "special" component types like DrawBoundary, FileUploadAndLabel etc that search results _are_ handling.

### CODE REVIEW

**Review by DafyddLlyr — APPROVED** [review]
Approved from a code POV - agree there's a bigger design question here to be decided on.

From a search / design perspective - I wonder if exposing this would make Editors expect there's more of a visual link between search results and Graph nodes (e.g. would there be an expectation that the data fields be "highlighted" in the graph the same way a search card's headline is?). Personally, I'd be against this for a few reasons - 
 - Complexity and overhead!
 - It breaks "Search" out of the right hand panel and sort of makes it a globally scoped feature
 - There's not a 1-to-1 relationship between graph node and data value, so it's never going to perfectly marry up. Likewise, this feature wouldn't work once search looks at non-data fields
 
So in summary, I think it's a useful and valuable addition to the graph (maybe we can have it toggle-able?), but I'd vote to not have these values dynamically linked to search results.

> Does not display data fields (yet?) for any of the "special" component types like DrawBoundary, FileUploadAndLabel etc that search results are handling.

Some of these are fairly complex as there's not a 1-to-1 relationship, I think it's totally fair to not expose the complexity and just show the "high level" data field which most components have.

**Comment by jessicamcinchak** [comment]
Closing in favor of #3789 which introduces toggle button & React based "DataField" rather than scss like here
